mpas_analysis/ocean/histogram.py

```diff
@@ -0,0 +1,595 @@
+        # which will perform some common setup, including storing:
+        #   self.inDirectory, self.plotsDirectory, self.namelist, self.streams
+        #   self.calendar
+        super(PlotRegionHistogramSubtask, self).setup_and_check()
+
+        self.xmlFileNames = []
+        for var in self.variableList:
+            self.xmlFileNames.append(
+                f'{self.plotsDirectory}/{self.filePrefix}_{var}_'
+                f'{self.regionName}_{self.season}.xml')
+
+    def run_task(self):
+        """
+        Plots histograms of properties in an ocean region.
+        """
+
+        self.logger.info(f"\nPlotting {self.season} histograms for "
+                         f"{self.regionName}")
+
+        config = self.config
+        sectionName = self.sectionName
+
+        calendar = self.calendar
+
+        main_run_name = config.get('runs', 'mainRunName')
+
+        region_mask_filename = self.mpasMasksSubtask.maskFileName
+
+        ds_region_mask = xarray.open_dataset(region_mask_filename)
+
+        mask_region_names = decode_strings(ds_region_mask.regionNames)
+        region_index = mask_region_names.index(self.regionName)
+
+        ds_mask = ds_region_mask.isel(nRegions=region_index)
+        cell_mask = ds_mask.regionCellMasks == 1
+
+        if len(self.obsDicts) > 0:
+            obs_region_mask_filename = self.obsMasksSubtask.maskFileName
+            ds_obs_region_mask = xarray.open_dataset(obs_region_mask_filename)
+            mask_region_names = decode_strings(ds_region_mask.regionNames)
+            region_index = mask_region_names.index(self.regionName)
+
+            ds_obs_mask = ds_obs_region_mask.isel(nRegions=region_index)
+            obs_cell_mask = ds_obs_mask.regionMasks == 1
+
+        in_filename = get_unmasked_mpas_climatology_file_name(
+            config, self.season, self.componentName, op='avg')
+        ds = xarray.open_dataset(in_filename)
+        ds = ds.where(cell_mask, drop=True)
+
+        base_directory = build_config_full_path(
+            config, 'output', 'histogramSubdirectory')
+
+        if self.weightList is not None:
+            weights_filename = \
+                f'{base_directory}/{self.filePrefix}_{self.regionName}_' \
+                'weights.nc'
+            ds_weights = xarray.open_dataset(weights_filename)
+
+        if self.controlConfig is not None:
+            control_run_name = self.controlConfig.get('runs', 'mainRunName')
+            control_filename = get_unmasked_mpas_climatology_file_name(
+                self.controlConfig, self.season, self.componentName, op='avg')
+            ds_control = xarray.open_dataset(control_filename)
+            base_directory = build_config_full_path(
+                self.controlConfig, 'output', 'histogramSubdirectory')
+            control_region_mask_filename = f'{base_directory}/histogram_' \
+                f'{control_run_name}_{self.regionName}_mask.nc'
```

Comment:
Oops, I missed that these file names also need to be updated:
```suggestion
            control_region_mask_filename = \
                f'{base_directory}/{self.filePrefix}_{self.regionName}_mask.nc'
```

Reply:
I double-checked but have not yet run the control comparison because I was doing that on lcrc and I think those machines are still down for maintenance.

Reply:
I'm rerunning my tests now.

Reply:
I've tested the control comparison successfully. In the process, I made some small changes to the treatment of `weightList` config option since it's an empty list by default: https://github.com/MPAS-Dev/MPAS-Analysis/pull/917/commits/1ba4bbb43df8fd19ae5f66bed1d3a6e30434e8ce